evm/src/main/java/org/hyperledger/besu/evm/UInt256.java

```diff
@@ -318,6 +363,85 @@ public UInt256 mulMod(final UInt256 other, final UInt256 modulus) {
     return new UInt256(result, modulus.length);
   }
 
+  /**
+   * Bitwise AND operation
+   *
+   * @param other The UInt256 to AND with this.
+   * @return The UInt256 result from the bitwise AND operation
+   */
+  public UInt256 and(final UInt256 other) {
+    int[] result = new int[N_LIMBS];
+    result[0] = this.limbs[0] & other.limbs[0];
+    result[1] = this.limbs[1] & other.limbs[1];
+    result[2] = this.limbs[2] & other.limbs[2];
+    result[3] = this.limbs[3] & other.limbs[3];
+    result[4] = this.limbs[4] & other.limbs[4];
+    result[5] = this.limbs[5] & other.limbs[5];
+    result[6] = this.limbs[6] & other.limbs[6];
+    result[7] = this.limbs[7] & other.limbs[7];
+    int resultLength = nSetLimbs(result);
```

Comment:
In the current implementation, you don't necessarily have to do this operation, you could just set the length to N_LIMBS (or Math.min(this.length, other.length)). If that can lead to performance improvements.
But this is what we discussed the other time, do we want to optimise nSetLimbs with Arrays.mismatch and use it with little cost all the time ? Or do we keep this length interpretation ?
Similarly for other bitwise ops.